p2p/transport/webrtc/transport.go

```diff
@@ -105,7 +108,9 @@ type iceTimeouts struct {
 	Keepalive  time.Duration
 }
 
-func New(privKey ic.PrivKey, psk pnet.PSK, gater connmgr.ConnectionGater, rcmgr network.ResourceManager, opts ...Option) (*WebRTCTransport, error) {
+type ListenUDPFn func(network string, laddr *net.UDPAddr) (net.PacketConn, error)
+
+func New(privKey ic.PrivKey, psk pnet.PSK, gater connmgr.ConnectionGater, rcmgr network.ResourceManager, listenUDP ListenUDPFn, opts ...Option) (*WebRTCTransport, error) {
```

Comment:
Should we make this depend on `quicreuse.Connmgr` like webtransport and quic constructors?
I don't have a strong opinion here.

Reply:
I purposely did not since it's nicer to require the bare minimum interface we actually need. WebTransport and QUIC do need the connmgr, WebRTC only needs a function that returns a `net.PacketConn`.